### DISCUSSION TITLE
Update github page

### DISCUSSION THREAD

**Discussion by btomecsko** [community/community#136872] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

It's been a little over a year since I touched my personal site and am working on updating. However, I have a dilemma. The react app for my portfolio is on my main branch and the website that is live right now is on the gh-pages branch. The gh-pages branch is sending the right content to my site, but it does not have all the files. It's like it is encrypted. Does anyone know how I can publish the changes from my main branch to gh-pages?

**Comment by Yackens** [comment, marked as answer]
To update your GitHub Pages site with the content from your main branch, you can follow these steps:

1. Build Your React App
First, you need to build your React app. This will generate static files that you can deploy to the gh-pages branch.

```npm run build```

This will create a build directory with the production-ready static files.

2. Install gh-pages Package
If you haven't already, install the gh-pages package to help with deploying to GitHub Pages.

```npm install --save-dev gh-pages```

3. Update package.json
Add the following scripts to your package.json to automate deployment:

```
"scripts": {
  "predeploy": "npm run build",
  "deploy": "gh-pages -d build"
}
```

4. Deploy Your Site
Run the deploy script to push the contents of the build directory to the gh-pages branch.

`npm run deploy`

If it helped, don't forget to mark it as the correct answer 😄

**Comment by ebndev** [comment]
Hi @btomecsko, thanks for posting in GitHub Discussions! 

I've gone ahead and moved this post to our `Pages` category, as this topic may be more relevant there and might help someone give you a nudge in the right direction. 😁

In the meantime, please feel encouraged to explore existing discussions in the category and utilize the search in case other users may have run into the same issue!
